src/strategies/evm/comp.ts

```diff
@@ -17,22 +17,21 @@ export default function createCompStrategy(): Strategy {
       params: string,
       provider: Provider
     ): Promise<bigint> {
+      const compContract = new Contract(params, ICompAbi, provider);
       const votingStrategyContract = new Contract(
         strategyAddress,
         CompVotingStrtategyAbi,
         provider
       );
 
-      let block;
+      let votingPower;
       const storedBlock = await votingStrategyContract.timestampToBlockNumber(timestamp);
-      if (storedBlock.toNumber() > 1) block = storedBlock.toNumber();
-      else {
-        block = (await provider.getBlockNumber()) - 1;
+      if (storedBlock.toNumber() > 1) {
+        votingPower = await compContract.getPriorVotes(voterAddress, storedBlock.toNumber());
+      } else {
+        votingPower = await compContract.getCurrentVotes(voterAddress);
```

Comment:
This will be problematic if we always query voting power at latest block and not specific block, on many proposals the vp might be misleading. I think we should use only "getCurrentVotes" but query the voting power at a specific block, this is possible with full archive node (we only use full archive node) with: 
```suggestion
        votingPower = await compContract.getCurrentVotes(voterAddress, { blockTag: storedBlock.toNumber() });
```

Reply:
Error come from this, Comp tokens dont support query for latest block
https://github.com/compound-finance/compound-protocol/blob/master/contracts/Governance/Comp.sol#L190

Reply:
> This will be problematic if we always query voting power at latest block and not specific block, on many proposals the vp might be misleading.

When querying for proposals we will still query at specific block: https://github.com/snapshot-labs/sx.js/blob/e1d51fbd3270c3612da5af43a620a8be13803404/src/strategies/evm/comp.ts#L28-L30

This PR only changes what happens when querying for timestamp that doesn't have block mapping (so no proposal with snapshot there).

> Comp tokens dont support query for latest block

We subtract -1 though, it seems that there is some timing issue where API returns higher block number than the contract sees, using `getCurrentVotes` eliminates that.

Reply:
You can use `getCurrentVotes` without specifying a blockTag for proposals and `getCurrentVotes` with a blockTag for single proposal